packages/teleport-plugin-common/src/node-handlers/node-to-jsx/utils.ts

```diff
@@ -147,20 +147,33 @@ export const createDynamicValueExpression = (
   params?: JSXGenerationParams
 ) => {
   const { projectContexts = {} } = params || {}
-  const refType = identifier.content.referenceType as 'prop' | 'state' | 'local' | 'ctx'
```

Comment:
this type cast does not guarantee we don't call this function without the other types of dynamic values. Better throw errors instead of assuming we know how to use the function.